tests/ignite/handlers/test_state_param_scheduler.py

```diff
@@ -0,0 +1,176 @@
+
+
+def test_linear_scheduler_linear_increase():
+    # Testing linear increase
+    engine = Engine(lambda e, b: None)
+    linear_state_parameter_scheduler = LinearStatePScheduler(
+        param_name="linear_scheduled_param", initial_value=0, step_constant=2, step_max_value=5, max_value=10,
+    )
+    linear_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+    engine.run([0] * 8, max_epochs=3)
+    torch.testing.assert_allclose(getattr(engine.state, "linear_scheduled_param"), 3.333333, atol=0.001, rtol=0.0)
+
+
+def test_linear_scheduler_max_value():
+    # Testing max_value
+    engine = Engine(lambda e, b: None)
+    linear_state_parameter_scheduler = LinearStatePScheduler(
+        param_name="linear_scheduled_param", initial_value=0, step_constant=2, step_max_value=5, max_value=10,
+    )
+    linear_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+    engine.run([0] * 8, max_epochs=10)
+    torch.testing.assert_allclose(getattr(engine.state, "linear_scheduled_param"), 10)
+
+
+def test_exponential_scheduler():
+    engine = Engine(lambda e, b: None)
+    exp_state_parameter_scheduler = ExpStatePScheduler(param_name="exp_scheduled_param", initial_value=10, gamma=0.99)
+    exp_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+    engine.run([0] * 8, max_epochs=2)
+    torch.testing.assert_allclose(getattr(engine.state, "exp_scheduled_param"), 10 * 0.99 * 0.99)
+
+
+def test_step_scheduler():
+    engine = Engine(lambda e, b: None)
+    step_state_parameter_scheduler = StepStatePScheduler(
+        param_name="step_scheduled_param", initial_value=10, gamma=0.99, step_size=5
+    )
+    step_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+    engine.run([0] * 8, max_epochs=10)
+    torch.testing.assert_allclose(getattr(engine.state, "step_scheduled_param"), 10 * 0.99 * 0.99)
+
+
+def test_multistep_scheduler():
+    engine = Engine(lambda e, b: None)
+    multi_step_state_parameter_scheduler = MultiStepStatePScheduler(
+        param_name="multistep_scheduled_param", initial_value=10, gamma=0.99, milestones=[3, 6],
+    )
+    multi_step_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+    engine.run([0] * 8, max_epochs=10)
+    torch.testing.assert_allclose(getattr(engine.state, "multistep_scheduled_param"), 10 * 0.99 * 0.99)
+
+
+def test_custom_scheduler():
+
+    initial_value = 10
+    gamma = 0.99
+    engine = Engine(lambda e, b: None)
+    lambda_state_parameter_scheduler = LambdaStatePScheduler(
+        param_name="custom_scheduled_param", lambda_fn=lambda event_index: initial_value * gamma ** (event_index % 9),
+    )
+    lambda_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+    engine.run([0] * 8, max_epochs=2)
+    torch.testing.assert_allclose(getattr(engine.state, "custom_scheduled_param"), 10 * 0.99 * 0.99)
+
+
+def test_simulate_and_plot_values():
```

Comment:
Let's use pytest parametrize here to setup the class : LinearStatePScheduler, etc